GoogleForCreators/web-stories-wp
# Enable new font picker + add Karma tests

## Summary



## Relevant Technical Choices



## To-do

- [x] Write QA instructions
- [x] Karma tests



## User-facing changes



## Testing Instructions
Recently used fonts:
I
1. Open a Story (existing or new)
2. Add a Text element and open the Font picker
3. Verify that there are no recently used fonts in the beginning of the fonts list
4. Now change the font family of the Text 5 times, every time a different font
5. Verify that each time after choosing the font, the font is added to the top of the fonts list, until there are all the 5 fonts.
6. Now use a new font family again
7. Verify that there are still only 5 fonts in the recently used fonts and the latest is on the top of the list

II
1. Open a Story (existing or new)
2. Add a Text element and open the Font picker
3. Switch the fonts between 2 different fonts a few times
4. Verify that both of these fonts are added to the recently used fonts but are never duplicated.

Curated list:
I
1. Add a Text element and open the Font picker
2. Scroll to the end of the font list and verify that the font picker only displays [the curated list of fonts](https://github.com/google/web-stories-wp/issues/1989#issuecomment-662253222) and not all the 1012 fonts.
3. Now start typing in "Princess" allowing the search to update after typing in each character, verify that new fonts appear including "Princess Sofia" which was not part of the list before.

Search:
I
1. Add a Text element and open the Font picker
2. Type in one character, e.g. "p". Verify that the list of fonts does not change when less than 2 characters are typed in.
3. Now type in more than 1 character, e.g. "pr" and verify that the list of fonts starts including new fonts, correctly updating the results based on what's written in the search.
4. Remove the typed in characters and verify that the search will reset to the original curated list again.

II
1. Add a Text element and open the Font picker
2. Choose a font, e.g. "Princess Sofia" (need to search for it first since it's not part of the curated list).
3. Once that's chosen, open the font picker again and type in "sof". Verify that "Princess Sofia" font will appear both in the recently used fonts and also in the search results, both with a checkmark in front (as the font is chosen).

III
1. Verify that when typing into the search, an "X" appears for clearing out the search.
2. Verify that clicking on it actually clears out search and restores the original curated list.

General:
I
1. Verify that when scrolling in the font list, the fonts are updated to display the real preview of the font.
2. Verify that the preview matches what's displayed in the editor after choosing the font.

II
1. Verify that keyboard can be used to move up-down in the font list and a font can be chosen by "Enter"/"Space" (open the font picker and use arrow keys to move up/down)
2. Verify that "Esc" closes the font picker.




---



Fixes #2418
